Presentation.Web/app/helpers/select2-option-format-helper.ts

```diff
@@ -27,18 +27,45 @@
         }
 
         public static formatIndentation(result: Models.ViewModel.Generic.Select2OptionViewModelWithIndentation<any>): string {
-            function visit(text: string, indentationLevel: number): string {
-                if (indentationLevel <= 0) {
-                    return text;
-                }
-                //indentation is four non breaking spaces
-                return visit("&nbsp&nbsp&nbsp&nbsp" + text, indentationLevel - 1);
+            const formattedResult = Select2OptionsFormatHelper.visit(result.text, result.indentationLevel);
+            return formattedResult;
+        }
+
+        public static formatIndentationWithUnitOrigin(result: Models.ViewModel.Generic.Select2OptionViewModelWithIndentation<Models.Api.Organization.OrganizationUnit>): string {
+            let isKitosUnit = true;
+            if (result.optionalObjectContext.externalOriginUuid) {
+                isKitosUnit = false;
             }
 
-            const formattedResult = visit(result.text, result.indentationLevel);
+            const formattedResult = Select2OptionsFormatHelper.visitUnitWithOrigin(result.text, result.indentationLevel, isKitosUnit);
             return formattedResult;
         }
 
+        private static visit(text: string, indentationLevel: number): string {
+            if (indentationLevel <= 0) {
+                return text;
+            }
+            //indentation is four non breaking spaces
+            return Select2OptionsFormatHelper.visit("&nbsp&nbsp&nbsp&nbsp" + text, indentationLevel - 1);
+        }
+
+        private static visitUnitWithOrigin(text: string, indentationLevel: number, isKitosUnit: boolean, indentationText: string = ""): string {
+            if (indentationLevel <= 0) {
+                return Select2OptionsFormatHelper.formatIndentationWithOriginText(text, indentationText, isKitosUnit);
+            }
+
+            //indentation is four non breaking spaces
+            return Select2OptionsFormatHelper.visitUnitWithOrigin(text, indentationLevel - 1, isKitosUnit, indentationText + "&nbsp&nbsp&nbsp&nbsp");
+        }
+
+        private static formatIndentationWithOriginText(text: string, indentationText: string, isKitosUnit: boolean) {
+            if (isKitosUnit) {
```

Comment:
some repetition in this method.. only one class differs in the two different html blocks.. no need to repeat everything